### PR TITLE
Proposal to make MapTypePlugin little filters more widely available

### PR DESCRIPTION
**Pull request by bobular**
I've added a `getLittleFilters` method to the map type plugin so that the implied little filters that a map type creates can be retrieved as actual filters.

It's partially implemented for donut map mode as a proof-of-concept.  When a string categorical variable is chosen, we will add a new filter to the timeslider data request in the following ways
1. filter on `variable in [ALL_VALUES]` when no values have been user-selected in the marker config table, or of "All other values" is in effect
2. filter on `variable in [USER_SELECTED_VALUES]` in all other cases.

We would need to do something for other variable types, and also implement for bubble numerator/denominator configs.

Is this a good way to go about this?

The places that need "explicit little filters" are
* timeslider and floating viz https://github.com/VEuPathDB/web-monorepo/issues/596 (UX debate needed maybe)
* when requesting sub-study counts in the new header in https://github.com/VEuPathDB/web-monorepo/issues/696
* also in #696 when requesting the list of sub-studies to display in the floater

In future possibly also https://github.com/VEuPathDB/web-monorepo/issues/548

### CODE REVIEW

**Comment by bobular** [comment]
Timeslider seems to "flash" (updates twice) when changing variables or checkbox selections for variables - need to look into this.

**Review by dmfalke — COMMENTED** [review]
To be honest, this approach makes me a little uneasy. It feels like too much child -> parent communication.

What about moving the `TimeSliderQuickFilter` rendering to each of the map type plugins? That keeps the flow of the props in a parent-> child direction.

File: packages/libs/eda/src/lib/map/analysis/TimeSliderQuickFilter.tsx [hunk 1]
```diff
@@ -39,13 +40,14 @@ export default function TimeSliderQuickFilter({
...
   filters,
+  littleFilters,
   starredVariables,
   toggleStarredVariable,
   config,
   updateConfig,
   siteInformation,
 }: Props) {
-  const findEntityAndVariable = useFindEntityAndVariable(filters); // filter sensitivity
+  const findEntityAndVariable = useFindEntityAndVariable(filters); // filter sensitivity only on main filters
```
[the remark this answers is not in the document]
Comment (bobular) [reply packages/libs/eda/src/lib/map/analysis/TimeSliderQuickFilter.tsx:50]: tbh that was just a hunch!

[the remark this answers is not in the document]
Comment (dmfalke) [reply packages/libs/eda/src/lib/map/analysis/TimeSliderQuickFilter.tsx:50]: I'm thinking it doesn't hurt to include it. I.e., it won't have an impact, unless the marker variable is the same as the time slider variable... which would be odd, I think.

My question is motivated by my desire to not expand the notion of "little filters" to other components. Couldn't we just concat filters and little filters into one array that gets passed to this component?

[the remark this answers is not in the document]
Comment (bobular) [reply packages/libs/eda/src/lib/map/analysis/TimeSliderQuickFilter.tsx:50]: Yes I think it's probably better not to alter the APIs of existing components (like I have done). I'll do a new prototype based on your feedback. Thanks!

[the remark this answers is not in the document]
Comment (bobular) [reply packages/libs/eda/src/lib/map/analysis/TimeSliderQuickFilter.tsx:50]: I was annoyed at not being able to use hooks in my approach here. It was setting some alarm bells ringing but not enough! Happy to try it another way.

**Comment by bobular** [comment]
Closing this as a dead end!

**Pull request closed without merging**